components/raftstore/src/store/worker/split_check.rs

```diff
@@ -335,6 +348,28 @@ where
             Task::ChangeConfig(c) => self.change_cfg(c),
             #[cfg(any(test, feature = "testexport"))]
             Task::Validate(f) => f(&self.cfg),
+            Task::GetRegionApproximateSizeAndKeys {
+                region,
+                pending_tasks,
+                cb,
+            } => {
+                if pending_tasks.fetch_sub(1, AtomicOrdering::SeqCst) > 1 {
```

Comment:
one region may be starving by this always be true

Reply:
This condition is before io operations. Starving can happen only if  task producing rate is greater than split checker processing(without io) rate. If heartbeat tick is 10 minutes，this will be hard to happen.

Reply:
@Connor1996  Do you have any suggestions?

Reply:
I think `pending_pd_heartbeat_tasks > 2` check in `heartbeat_pd` is enough, and a log there?

Reply:
If there are two heartbeats，skip the first can reduce one IO without any side effects. Maybe it is not a bad choice?

Reply:
Oh, it's a counter for every peer. I misunderstood the code.